proxies/shared/resources/jsc/Routing.CheckValid.js

```diff
@@ -0,0 +1,36 @@
+/*
+When a method is used to access an endpoint that doesn't support it, the API responds with a 405
+When a GET method is used to access an endpoint that doesn't exist, the API responds with a 404
+*/
+
+var is404 = true;
+var is405 = false;
+
+const requestMethod = context.getVariable("request.verb").toLowerCase();
+const requestPath = context.getVariable("proxy.pathsuffix");
+
+// does request path have ? args on it
+
+const validPaths = {
+    '/_ping' : [
+        'GET',
+        'HEAD'
+    ],
+    '/v1/message-batches' : [
+        'POST'
+    ],
+    '/v1/messages' : [
+        'POST'
+    ],
+    '/v1/messages/:messageId' : [
```

Comment:
As this supports `GET`, it should also allow `HEAD` to go through.

We should allow all `OPTIONS` requests to go through.

Reply:
The options requests may already be getting let through elsewhere so may not be relevant.